api/src/test/java/edu/cornell/mannlib/vitro/webapp/dynapi/RESTEndpointITest.java

```diff
@@ -0,0 +1,262 @@
+
+        return requests;
+    }
+
+    public static List<Object[]> requests(String resourceName, String individualUri, String customRestActionName) {
+        // would be ideal if sparql made actual call during integration tests
+        String sparqlResFile = "sparql/response/json/sparql-empty-success.json";
+
+        // TODO: create appropriate request body, request params, and response body files when applicable
+        // String restReqParamsFile = format("%s.json", resourceName);
+        // String restReqBodyFile = format("%s.json", resourceName);
+        // String restResBodyFile = format("%s.json", resourceName);
+        String restReqParamsFile = "test_collection_resource.json";
+        String restReqBodyFile = "test_collection_resource.json";
+        String restResBodyFile = "test_collection_resource.json";
+
+        String resourcePath = format("/1/%s", resourceName);
+
+        String individualResourcePath = format("%s/resource:%s", resourcePath, new String(Base64.getEncoder().encode(individualUri.getBytes())));
+
+        String customRestActionPath = format("%s/%s", individualResourcePath, customRestActionName);
+
+        return new ArrayList<>(Arrays.asList(new Object[][] {
+            { "POST",   resourcePath,           restReqParamsFile, restReqBodyFile, sparqlResFile, SC_OK,                 restResBodyFile, "Create collection resource" },
+            { "GET",    resourcePath,           restReqParamsFile, null,            sparqlResFile, SC_OK,                 restResBodyFile, "Get collection resources" },
+            { "PUT",    resourcePath,           restReqParamsFile, null,            null,          SC_METHOD_NOT_ALLOWED, null,            "Update not allowed on collection" },
+            { "PATCH",  resourcePath,           restReqParamsFile, null,            null,          SC_METHOD_NOT_ALLOWED, null,            "Patch not allowed on collection" },
+            { "DELETE", resourcePath,           restReqParamsFile, null,            null,          SC_METHOD_NOT_ALLOWED, null,            "Delete not allowed on collection" },
+
+            { "POST",   individualResourcePath, restReqParamsFile, null,            null,          SC_METHOD_NOT_ALLOWED, null,            "Create not allowed on individualt resource" },
+            { "GET",    individualResourcePath, restReqParamsFile, null,            sparqlResFile, SC_OK,                 restResBodyFile, "Get individual resource" },
+            { "PUT",    individualResourcePath, restReqParamsFile, restReqBodyFile, sparqlResFile, SC_OK,                 restResBodyFile, "Update individual resource" },
+            { "PATCH",  individualResourcePath, restReqParamsFile, restReqBodyFile, sparqlResFile, SC_OK,                 restResBodyFile, "Patch individual resource" },
+            { "DELETE", individualResourcePath, restReqParamsFile, null,            sparqlResFile, SC_OK,                 restResBodyFile, "Delete individual resource" },
```

Comment:
Do we need to add api/src/test/resources/dynapi/mock/rest/response/body/delete/test_collection_resource.json?

Reply:
Yeah eventually. They are currently not being read. Would you like I add them all end ensure they are read? Then when we can assert against them, we do.

Reply:
Yeah, added. Thanks.